src/main/java/bio/terra/service/dataset/flight/inheritsteward/SetInheritStewardFlight.java

```diff
@@ -42,20 +42,27 @@ public EnableInheritStewardFlight(FlightMap inputParameters, Object applicationC
     AuthenticatedUserRequest userReq =
         inputParameters.get(JobMapKeys.AUTH_USER_INFO.getKeyName(), AuthenticatedUserRequest.class);
 
-    boolean inheritSteward = true;
+    boolean inheritSteward =
+        inputParameters.get(DatasetWorkingMapKeys.INHERIT_STEWARD, Boolean.class);
     addStep(new LockDatasetStep(datasetService, datasetId, false));
     addStep(new SetInheritStewardFlagStep(datasetDao, datasetId, inheritSteward));
-    addStep(new GetSnapshotIdsStep(snapshotDao, datasetId));
-    addStep(new SetParentOnSnapshotsStep(iamService, datasetId, userReq));
-    addStep(new GetSnapshotGoogleProjectIdsStep(snapshotService, datasetId));
-    addStep(new SetAuthGcpUserRolesStep(resourceService, custodianEmail, inheritSteward));
+    addStep(new GetSnapshotIdsStep(snapshotDao, iamService, userReq, datasetId, inheritSteward));
+    addStep(new SetParentOnSnapshotsStep(iamService, datasetId, userReq, inheritSteward));
+    addStep(
+        new SetAuthGcpUserRolesStep(
+            resourceService, snapshotService, custodianEmail, inheritSteward));
     addStep(
         new SetAuthTabularAclStep(
             bigQuerySnapshotPdao, snapshotService, custodianEmail, inheritSteward));
     addStep(new AdjustStewardMembersStep(userReq, iamService, inheritSteward));
     addStep(new UnlockDatasetStep(datasetService, false));
     addStep(
         new JournalRecordUpdateEntryStep(
-            journalService, userReq, datasetId, IamResourceType.DATASET, "Enable inherit steward"));
+            journalService,
+            userReq,
+            datasetId,
+            IamResourceType.DATASET,
+            String.format(
+                "Set Inherit Steward flag to %s on dataset, %s", inheritSteward, datasetId)));
```

Comment:
We don't usually add the ID to the journal message if it's the same ID that the journal entry is for. The journal API is already keyed off of the object id (and type) so those attributes don't need to be added here.
```suggestion
            "Set Inherit Steward flag to %s".formatted(inheritSteward));
```